tests/units/cli/test_cli.py

```diff
@@ -48,6 +52,42 @@ def teardown_method(self):
         except Exception:
             pass
 
+    @staticmethod
+    def _init_for_sync_tables_states_cleanup(tables_arg: str = None) -> PipelineWise:
+        temp_path = '/tmp/pwtest/'
+        args = CliArgs(target='target_one', tap='tap_one', tables=tables_arg)
+        pipelinewise = PipelineWise(args, CONFIG_DIR, VIRTUALENVS_DIR)
+        pipelinewise.tap['files']['state'] = f'{temp_path}state.json'
+        pipelinewise.venv_dir = temp_path
+
+        # Making a test tap bin file
+        os.makedirs(f'{temp_path}pipelinewise/bin')
+        with open(f'{temp_path}pipelinewise/bin/mysql-to-snowflake', 'a', encoding='UTF-8'):
+            pass
+
+        return pipelinewise
+
+    @staticmethod
+    def _make_sample_state_file(test_state_file: str) -> None:
+        sample_state_data = {
+            'currently_syncing': None,
+            'bookmarks': {
+                'tap_id-table1': {'foo': 'bar'},
+                'tap_id-table2': {'foo': 'bar'},
+                'tap_id-table3': {'foo': 'bar'}
+            }
+        }
+        with open(test_state_file, 'w', encoding='UTF-8') as state_file:
+            json.dump(sample_state_data, state_file)
+
+    @staticmethod
+    def _assert_calling_sync_tables(pipelinewise: PipelineWise, side_effect_method: Optional[Callable] = None) -> None:
+        with patch('pipelinewise.cli.pipelinewise.PipelineWise.run_tap_fastsync') as mocked_fastsync:
+            if side_effect_method:
+                mocked_fastsync.side_effect = side_effect_method
+            pipelinewise.sync_tables()
+        mocked_fastsync.assert_called_once()
```

Comment:
it is not advised to use `assert_called_once`, it's a fake method that will always pass. Instead use `assert mocked_fastsync.call_count == 1`

Reply:
This is the implementation of `assert_called_once`:

```python
   def assert_called_once(self):
        """assert that the mock was called only once.
        """
        if not self.call_count == 1:
            msg = ("Expected '%s' to have been called once. Called %s times.%s"
                   % (self._mock_name or 'mock',
                      self.call_count,
                      self._calls_repr()))
            raise AssertionError(msg)
```

How do you believe it is a fake method and always passes? 
To test it: just add a `return` in `pipelinewise.cli.pipelinewise.sync_tables` just before `self.run_tap_fastsync` and you see the test fails because of not calling the method.

Reply:
https://engineeringblog.yelp.com/2015/02/assert_called_once-threat-or-menace.html

Reply:
1. In this article they are talking about 'mock' package, in pipelinewise we use the mock from `unittest`. They are different!
1. This article is written on Feb 4 2015! on that time the latest version of `mock` was `1.0.1` which had this bug. the latest version of `mock` is '4.0.3' which this issue is solved.

Reply:
hmmm ok